pkg/kubectl/cmd/auth/cani.go

```diff
@@ -196,6 +199,12 @@ func (o *CanIOptions) Validate() error {
 		if o.Resource != (schema.GroupVersionResource{}) || o.ResourceName != "" {
 			return fmt.Errorf("NonResourceURL and ResourceName can not specified together")
 		}
+	} else if !o.Resource.Empty() && !o.AllNamespaces && o.DiscoveryClient != nil {
+		if namespaced, err := isNamespaced(o.Resource, o.DiscoveryClient); err == nil && !namespaced {
+			fmt.Fprintf(o.ErrOut, "Warning: resource %s is not namespace scoped\n", o.Resource.Resource)
+		} else if err != nil {
+			fmt.Fprintf(o.ErrOut, "Warning: %v\n", err)
```

Comment:
don't print lookup errors... missing or unknown resources were already warned about and we don't want double warning messages